src/agent/src/rpc.rs

```diff
@@ -1628,34 +1360,33 @@ fn find_process<'a>(
     Ok(p)
 }
 
-pub fn start<S: Into<String>>(sandbox: Arc<Mutex<Sandbox>>, host: S, port: u16) -> Server {
-    let env = Arc::new(
-        EnvBuilder::new()
-            .cq_count(1)
-            .wait_thread_count_default(5)
-            .wait_thread_count_min(1)
-            .wait_thread_count_max(10)
-            .build(),
-    );
-    let worker = agentService {
-        sandbox: sandbox,
+pub fn start<S: Into<String>>(s: Arc<Mutex<Sandbox>>, host: S, port: u16) -> ttrpc::Server {
+    let agent_service = Box::new(agentService {
+        sandbox: s,
         test: 1,
-    };
-    let service = protocols::agent_grpc::create_agent_service(worker);
-    let hservice = protocols::health_grpc::create_health(healthService);
-    let mut server = ServerBuilder::new(env)
-        .register_service(service)
-        .register_service(hservice)
-        .requests_slot_per_cq(1024)
-        .bind(host, port)
-        .build()
-        .unwrap();
-    server.start();
-    info!(sl!(), "gRPC server started");
-    for &(ref host, port) in server.bind_addrs() {
-        info!(sl!(), "listening"; "host" => host,
-        "port" => port);
-    }
+    }) as Box<dyn protocols::agent_ttrpc::AgentService + Send + Sync>;
+
+    let agent_worker = Arc::new(agent_service);
+
+    let health_service =
+        Box::new(healthService {}) as Box<dyn protocols::health_ttrpc::Health + Send + Sync>;
+    let health_worker = Arc::new(health_service);
+
+    let aservice = protocols::agent_ttrpc::create_agent_service(agent_worker);
+
+    let hservice = protocols::health_ttrpc::create_health(health_worker);
+
+    let mut addr: String = host.into();
+    addr.push_str(":");
+    addr.push_str(&port.to_string());
+
+    let server = ttrpc::Server::new()
+        .bind(addr.as_str())
+        .unwrap()
+        .register_service(aservice)
+        .register_service(hservice);
+
+    info!(sl!(), "TTRPC server started");
```

Comment:
OOI, is the official project name "ttrpc", "TTRPC" or "ttRPC"? I'd guess "ttRPC" as that seems to align most closely to "gRPC"?

Reply:
changed it to "ttRPC" .